aten/src/ATen/templates/RegisterDispatchKey.cpp

```diff
@@ -37,17 +37,23 @@
 
 namespace at {
 
-${dispatch_definitions}
-
 // NB: TORCH_LIBRARY_IMPL must be in an anonymous namespace to avoid
 // ambiguity with conflicting identifiers that may have been defined in
 // at namespace already.
 namespace {
 
+${dispatch_anonymous_definitions}
+
 TORCH_LIBRARY_IMPL(aten, ${DispatchKey}, m) {
   ${dispatch_registrations}
 }
 
 } // anonymous namespace
 
+namespace ${dispatch_namespace} {
+
+${dispatch_namespaced_definitions}
+
+} // namespace ${dispatch_namespace}
+
 } // namespace at
```

Comment:
nit: change `namespace at` to `namespace {dispatch_namespace}`?

Reply:
but... it is the closing namespace for at? :o)

Reply:
yup ignore me 😄